<!-- camera.launch -->
<!-- Usage: "roslaunch igvc camera.launch camera:=[left,right,center]" -->
<launch>
	<arg name="camera" default="center" />
	
	<group if="$(eval 'center' in arg('camera'))">
		<node name="usb_cam_center" pkg="usb_cam" type="usb_cam_node">
		    <param name="path" type="string" value="file://$(find igvc)/../../../"/>
		    <param name="video_device" type="string" value="/dev/igvc_usb_cam_center" />
		    <param name="pixel_format" type="string" value="yuyv" />
		    <param name="camera_frame_id" type="string" value="/usb_cam_center" />
	 	    <param name="camera_info_url" type="string" value="file://$(find igvc)/../sandbox/camera_config/usb_cam_center.yaml" />
		    <param name="camera_name" type="string" value="usb_cam_center" />
		</node>
	</group>

	<group if="$(eval 'right' in arg('camera'))">
		<node name="usb_cam_right" pkg="usb_cam" type="usb_cam_node">
		    <param name="path" type="string" value="file://$(find igvc)/../../../"/>
		    <param name="video_device" type="string" value="/dev/igvc_usb_cam_right" />
		    <param name="pixel_format" type="string" value="yuyv" />
		    <param name="camera_frame_id" type="string" value="/usb_cam_right" />
	 	    <param name="camera_info_url" type="string" value="file://$(find igvc)/../sandbox/camera_config/usb_cam_right.yaml" />
		    <param name="camera_name" type="string" value="usb_cam_right" />
		</node>
	</group>

	<group if="$(eval 'left' in arg('camera'))">
		<node name="usb_cam_left" pkg="usb_cam" type="usb_cam_node">
		    <param name="path" type="string" value="file://$(find igvc)/../../../"/>
		    <param name="video_device" type="string" value="/dev/igvc_usb_cam_left" />
		    <param name="pixel_format" type="string" value="yuyv" />
		    <param name="camera_frame_id" type="string" value="/usb_cam_left" />
	 	    <param name="camera_info_url" type="string" value="file://$(find igvc)/../sandbox/camera_config/usb_cam_left.yaml" />
		    <param name="camera_name" type="string" value="usb_cam_left" />
		</node>
	</group>


	
</launch>
